File: .3rd_party/github.com/jackgene/elm-syntax-highlight/demo/examples/syntax.go
package main

import (
	"fmt"
)

// Some values
const Pi float64 = float64(3.14)
var chr byte = 'x'
var num float64 = 42
var nums []int = []int{1, 2, 3}
var flag bool = true
var dict map[int]string = nil

func init() {
	num := Pi
	iceCream := "vanilla"
	fmt.Println(num, nums, iceCream)
	// Code snippet
	if iceCream == "chocolate" {
		fmt.Printf(`Yay, I love %s ice cream!`, iceCream)
	} else {
		fmt.Printf("Awwww, but chocolate is my favorite...")
	}
}

// Function declaration
func Multiply(num1, num2 float64) float64 {
	result := num1 * num2 * 1.0
	return result
}

// Function evaluation
var product float64 = Multiply(float64(nums[0]), 10)

// Type definition
type Shape interface {
	Size() float64
	Fits(Shape) bool
}
type Thing struct{}
type Polygon struct {
	Thing
	name   string
	Height float64
	Width  float64
}

func (p Polygon) GetName() string {
	return p.name
}

func (p Polygon) Size() float64 {
	return Multiply(p.Height, p.Width)
}

func (p Polygon) Fits(in Shape) bool {
	return p.Size() < in.Size()
}

// Function using declared type
func MakePolygon(height float64) Polygon {
	return Polygon{name: "", Height: height, Width: product}
}
var s Shape = MakePolygon(1.4)
func PrintPolygon() {
	p := s.(Polygon)
	fmt.Println(p.GetName(), p.Height + p.Width)
}

// Enum declaration
type Direction byte
const (
	NORTH Direction = iota
	SOUTH
	EAST
	WEST
)
func ReadDirection() {
	println()
	fmt.Println(NORTH)
}

File: .3rd_party/github.com/jackgene/elm-syntax-highlight/demo/examples/syntax.swift
import Foundation

// Some values
let Pi: Float = 3.14 as Float
var byte: UInt8? = nil
var num: Double = 42
let nums: [UInt] = [1, 2, 3]
let flag: Bool = true
let iceCream: String = flag ? "chocolate" : "vanilla"

// Code snippet
if iceCream == "chocolate" {
    print("Yay, I love \(iceCream) ice cream!")
} else {
    print("Awwww, but chocolate is my favorite...")
}

// Function declaration
fileprivate func multiply(_ num1: Double, _ num2: Double) -> Double {
    let result = num1 * num2 * 1.0
    return result
}

// Function evaluation
let product: Double = multiply(num, Double(10))

// Type definition
public protocol Shape {
    var size: Double { get }
    func fits(within: Shape) -> Bool
}
public class Thing {
    internal let name: String

    public init(name: String) {
        self.name = name
    }

    public func getName() -> String {
        name
    }
}
internal class Polygon: Thing {
    let height: Double
    let width: Double

    @objc
    public init(height: Double, width: Double) {
        self.height = height
        self.width = width
        super.init(name: "Polygon")
    }

    public override func getName() -> String {
        print("hello");
        let _ = 1 * 2
        return super.name
    }

    var area: Double {
        get {
            return multiply(self.height, self.width)
        }
    }
}
extension Polygon: Shape {
    var size: Double {
        get {
            return self.area
        }
    }

    public func fits(within: Shape) -> Bool {
        self.size < within.size
    }
}

// Function using declared type
func makePolygon(height: Double) -> Polygon {
    return Polygon(height: height, width: product)
}
let s: Shape = makePolygon(height: 1.4)
print(s is [String:Polygon])
guard let p = s as? Polygon else {
    exit(1)
}
print("\(p.getName()) \(p.height + p.width)")

// Enum declaration
enum Direction {
    case north
    case south
    case east
    case west
}
print(Direction.north)
